### ISSUE TITLE
Add `ref` for `highway=emergency_access_point` and `emergency=access_point`

### ISSUE THREAD

**Issue by mcliquid** [streetcomplete/StreetComplete#4384]
### General
Affected tag(s) to be modified/added:
[highway=emergency_access_point](https://wiki.openstreetmap.org/wiki/Tag:highway%3Demergency_access_point)
[highway=access_point](https://wiki.openstreetmap.org/wiki/Tag:emergency%3Daccess_point)
Question asked: **Is there a reference number on this (emergency) access point?**

### Checklist
Checklist for quest suggestions (see [guidelines](https://github.com/streetcomplete/StreetComplete/blob/master/QUEST_GUIDELINES.md)):
- [x] 🚧 To be added tag is established and has a useful purpose
- [x] 🤔 Any answer the user can give must have an equivalent tagging (Quest should not reappear to other users when solved by one)
"Answer: no" would add [noref=yes](https://wiki.openstreetmap.org/wiki/Key:noref)
- [x] 🐿️ Easily answerable by any pedestrian from the outside but a survey is necessary
Regarding wiki, all access points have a number on the sign.
- [x] 💤 Not an overwhelming percentage of quests have the same answer (No spam)
There are "only" around 1400 missing ref's in Germany.
- [x] 🕓 Applies to a reasonable number of map data (Worth the effort)
Probably not really worth, but there a lot of similar quests which could be duplicated for this case.
Regarding taginfo there are only 4-5 % of all access points without a ref-number which are worldwide 2.342 issues for this quest.

**Disclaimer: I'm perfectly fine with closing the issue if it doesn't make sense.**

### Ideas for implementation

**Element selection:** 
(highway=emergency_access_point or emergency=access_point) and ref!=*
Overpass Turbo for Germany: https://overpass-turbo.eu/s/1lVd

**UI:**
UI and behaviour could be identical with the bus ref quest or the post_box ref quest.

**Comment by matkoniecz** [comment]
emergency access point will have code almost by definition, otherwise there are useless, right?

>  Regarding taginfo there are only 4-5 % of all access points without a ref-number which are worldwide 2.342 issues for this quest.

2132 after limiting to nodes: https://overpass-turbo.eu/s/1lVf

A bit borderline, but this data can be actually useful (for emergency services using OSM data and for resurvey whether emergency points are actually signed).

**Comment by HolgerJeromin** [comment]
A quest should have as "other" option:
* no, this is a assembly_point https://wiki.openstreetmap.org/wiki/Tag:emergency=assembly_point

**Comment by westnordost** [comment]
- Will the surveyor know what to look for when asked for an "emergency access point"?
- Similarly, will he know if something is an "assembly point"?

**Comment by HolgerJeromin** [comment]
> Will the surveyor know what to look for when asked for an "emergency access point"?

In my opinion: yes
At least if they find them 

> Similarly, will he know if something is an "assembly point"?

I know them in germany with always the same sign. Other countries?
But as this an menu item we can not have an icon direct visible.

**Comment by westnordost** [comment]
That's cool, congratulations! A few remarks of what you showed so far:
1. Title question does not need to mention "emergency access point" because it is already mentioned in the hint below
2. I guess if the emergency access point has a `name` (or anything else that could be used otherwise to reference it), it also should not be asked
3. It probably needs a "does not exist here" option? Though, this is a bit dangerous of course, if the emergency access point looks different than one would expect

Regarding your question: See for example `SmoothnessAnswer` on how to add "out of the line" answer options.

Regarding the eligibility of this quests, my concerns are:
- with `noref=yes`, `name` and `ref` excluded, how many elements are these exactly worldwide? Is this even worth the effort then? (A number has been cited earlier but I think it did not take `name` and `noref` into account)
- not sure if "the average Joe" would be able to identify an emergency access point as one? Maybe a picture can help - if they don't look very different across countries i.e. their appearance is somewhat international. Needs some research
- not sure if "the average Joe" would know what an "assembly point" is. Maybe a picture can help, i.e. when the user taps on "is assembly point", a picture is shown, asking "so something like this is visible?". (The same is done for when the user answers that something is a living street in the maxspeed quest). That too, depends on whether there is some kind of international symbol maybe?

**Comment by westnordost** [comment]
> I can't get it to work yet, this seems to be too complex for me at the moment, as a lot of references to other functions have to be created. For this I would have to read up more deeply but I will do some more attempts to add more functionality.

add
`override val isDeleteElementEnabled = true`

**Comment by matkoniecz** [comment]
@mcliquid You can check tag value - for example in `AddTactilePavingKerb`

```
    override fun applyAnswerTo(answer: Boolean, tags: Tags, timestampEdited: Long) {
        tags.updateWithCheckDate("tactile_paving", answer.toYesNo())
        if (tags["kerb"] != "no") {
            tags["barrier"] = "kerb"
        }
    }
```

**Comment by westnordost** [comment]
The ticket will be closed automatically after the PR has been merged. Since the PR includes new string, this will happen in preparation of the v48.0 release, latest on the 30th of September.

**Comment by mnalis** [comment]
Just to clarify @westnordost, will the strings for translation be available on 30th of September, or would `48.0` itself be released on 30th, and translation are supposed to be finished before that?

**Comment by matkoniecz** [comment]
@mnalis As I understand 30th of September is planned to be date of alpha release and date when strings become available for translations (and initial step in v48 release cycle)

**Comment by mnalis** [comment]
Oh, OK, so `48.0-alpha1` gets released on 2022-09-30 along with translation strings, and actual `48.0` releases some weeks after that. 

Although as a translator I wouldn't mind at all if translation strings were released before `48.0-alpha1` (so if I'm quick I might get a chance to test my strings live in `48.0-alpha1`, and maybe then tweak them a little for `48.0`). But not a critical thing, just would be nice.
